internal/server/adapters/store/memory/store.go

```diff
@@ -0,0 +1,125 @@
+package memory
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/VoevodinAnton/metrics/internal/server/models"
+)
+
+var (
+	ErrMetricNotFound = errors.New("metric not found")
+)
+
+type Store struct {
+	gaugeMetrics   sync.Map
+	counterMetrics sync.Map
+	sync.Mutex
```

Comment:
Why do you need it?

Reply:
Writing to the map happens here:
GetGaugeMetrics(ctx context.Context)
GetCounterMetrics(ctx context.Context) methods